client/app/scripts/charts/edge.js

```diff
@@ -5,6 +5,30 @@ import classNames from 'classnames';
 import { enterEdge, leaveEdge } from '../actions/app-actions';
 import { encodeIdAttribute, decodeIdAttribute } from '../utils/dom-utils';
 
+function IsStorageComponent(id) {
+  if (id === 'persietent_volume' || id === 'storage_class' || id === 'persistent_volume_claim') {
+    return true;
+  }
+  return false;
+}
+
+// getAdjacencyClass takes id which contains information about edge as a topology
+// of parent and child node.
+
+function getAdjacencyClass(id) {
+  const topologyId = id.split('---');
+  const from = topologyId[0].split(';');
+  const to = topologyId[1].split(';');
+  if (from[1] !== undefined && to[1] !== undefined) {
+    from[1] = from[1].slice(1, -1);
+    to[1] = to[1].slice(1, -1);
```

Comment:
Assigning into the array here is still very confusing to me.

It would be clearer to introduce some new variables instead, something like:
```js
const fromStorageType = from[1].slice(1, -1);
const toStorageType = to[1].slice(1, -1);
```